sources/metricdog/src/metricdog.rs

```diff
@@ -156,3 +157,29 @@ impl Metricdog {
         Ok(())
     }
 }
+
+/// reqwest needs a URL protocol scheme to be present, but other implementations assume
+/// `http://` as the scheme when none is present. We need to check the `HTTPS_PROXY` env and
+/// reset it with `http://` when no scheme is present.
+// TODO - remove this if reqwest adds support for proxy addresses with no scheme.
+fn fix_https_proxy_env() {
```

Comment:
Also as I asked on #1324, do we need to fix up the no_proxy env as well?

Reply:
> Also as I asked on #1324, do we need to fix up the no_proxy env as well?

Same answer here in case someone doesn't see the other PR. For NO_PROXY values, testing proved that reqwest does not require the scheme (whereas it does need it for the HTTPS_PROXY value).